src/utils/executeInSeries.js

```diff
@@ -9,19 +9,12 @@
  * @param {Object|null} context - The object the functions must be bound to
  * @param {Function} done - The final callback to execute once all functions are executed
  */
-export default (callbacks, args, context, done) => {
-    const cbs = [...callbacks]; // clone the array to avoid modifying the original
-
-    const iterator = (iteratorCallbacks, iteratorArgs) => {
-        if (!iteratorCallbacks.length) {
-            return typeof done === 'function' ? done() : true;
+export default (callables, args, context) =>
+    new Promise(async (resolve, reject) => {
```

Comment:
I think we can do better with Promise.allSettled/Promise.race (https://2ality.com/2019/08/promise-combinators.html). Maybe Promise.any (https://2ality.com/2019/12/promise-any.html)

We can keep this way and improve after the general review